src/palette.ml

```diff
@@ -19,6 +19,71 @@ let generate_plasma_palette (size : int) : t =
 
     Int32.of_int (((int_of_float fred) * 65536) + ((int_of_float fgreen) * 256) + (int_of_float fblue))
   )
+  
+let generate_vapourwave_palette (size : int) : t =
+  if size <= 0 then raise (Invalid_argument "Palette size must not be zero negative");
+  let color1 = 0x7f3b8f (* Pastel purple *)
+  and color2 = 0x80cfcf (* Pastel cyan *) in
+  
+  Array.init size (fun index ->
+    let ratio = float_of_int index /. float_of_int (size - 1) in
+      
+    let red1 = (color1 / 65536) land 0xFF in
```

Comment:
These constants don't depend on index, so can be pulled out of the loop, which saves on unnecessary recalculations.